app/controllers/api/functions.php

```diff
@@ -442,6 +448,25 @@
         ]), Response::MODEL_RUNTIME_LIST);
     });
 
+App::get('/v1/functions/specs')
+    ->groups(['api', 'functions'])
+    ->desc('Get available function specs')
+    ->label('scope', 'functions.read')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'functions')
+    ->label('sdk.method', 'getSpecs')
+    ->label('sdk.description', '/docs/references/functions/get-specs.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_SPECS)
+    ->inject('response')
+    ->action(function (Response $response) {
+        $response->dynamic(new Document([
+            'memory' => Memory::getAllowedValues(),
+            'cpus' => Cpus::getAllowedValues()
+        ]), Response::MODEL_SPECS);
+    });
```

Comment:
as discussed, we should probably store the list of all sizes as a `Config` or something. And that way we can reuse the config inside the validators & the response here.

Then we can use the `plan` object to store the whitelist of allowed sizes for each tier. 
https://github.com/appwrite/appwrite/blob/main/app/init.php#L1698-L1700